Content.Shared/StepTrigger/Systems/ShoesRequiredStepTriggerSystem.cs

```diff
@@ -19,18 +18,33 @@ public override void Initialize()
 
     private void OnStepTriggerAttempt(EntityUid uid, ShoesRequiredStepTriggerComponent component, ref StepTriggerAttemptEvent args)
     {
-        if (_tagSystem.HasTag(args.Tripper, "ShoesRequiredStepTriggerImmune"))
+        // checks if the entity itself is the one with the component
+        if (HasComp<ShoesRequiredStepTriggerImmuneComponent>(args.Tripper))
         {
             args.Cancelled = true;
             return;
         }
 
+        // if not, checks the inventory
         if (!TryComp<InventoryComponent>(args.Tripper, out var inventory))
             return;
 
-        if (_inventory.TryGetSlotEntity(args.Tripper, "shoes", out _, inventory))
+        // early exit if no shoe slot exists at all, dionas suffer
+        // remove to give dionas their freedom from glass
+        if (!_inventory.HasSlot(args.Tripper, "shoes", inventory))
+            return;
+
+        // go through all equipped items, checks if item is equipped in shoe slot or has shoe immmune component
+        if (_inventory.TryGetContainerSlotEnumerator(args.Tripper, out var containerSlotEnumerator))
         {
-            args.Cancelled = true;
+            while (containerSlotEnumerator.NextItem(out var item, out var slot))
```

Comment:
you need to check that these items are actually in their appropriate item slots and not in pockets or something. I think this is done with relay events most of the time, so you should look into this.

Reply:
okay, how about this...

Reply:
I'll look through the relay events now

Reply:
If you want me to use relay events then I'll likely have to check at the time anything is equipped and check if anything has the comp and save that status. That way the player's entire inventory isn't being checked every time they slip, but instead only when their equipped item changes. Will work on that...

Reply:
Currently I re-implemented it to just check every slot besides pockets...

Reply:
I think we have relay events for specific slots now?